packages/babel-plugin/src/utils/traverse-expression-with-evaluation.tsx

```diff
@@ -0,0 +1,162 @@
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+const traverseMemberExpression = (expression: t.MemberExpression, meta: Metadata) => {
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  const { accessPath, bindingIdentifier } = getMemberExpressionMeta(expression);
+  const binding = updatedMeta.parentPath.scope.getBinding(bindingIdentifier.name);
+  const resolvedBinding = resolveBindingNode(binding, updatedMeta);
+
+  if (resolvedBinding && resolvedBinding.constant && t.isObjectExpression(resolvedBinding.node)) {
+    const objectValue = getValueFromObjectExpression(
+      resolvedBinding.node,
+      accessPath
+    ) as t.Expression;
+    // We recursively call get interpolation until it not longer returns an identifier or member expression
+    ({ value, meta: updatedMeta } = traverseExpressionWithEvaluation(
+      objectValue,
+      resolvedBinding.meta
+    ));
+  }
+
+  return createResultPair(value as t.Expression, updatedMeta);
+};
+
+/**
+ * Will look in an expression and return the actual value along with updated metadata.
+ *
+ * E.g: If there was a function called `size` that is set somewhere as
+ * `const size = () => 10` or `const size = function() { return 10; }` or `function size() { return 10; }`,
+ * passing the `size` identifier to this function would return `10` (it will recursively evaluate).
+ *
+ * @param expression Expression we want to interrogate.
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+const traverseFunction = (expression: t.Function, meta: Metadata) => {
+  let value: t.Node | undefined | null = undefined;
+  let updatedMeta: Metadata = meta;
+
+  if (t.isBlockStatement(expression.body)) {
+    traverse(expression.body, {
+      noScope: true,
+      ReturnStatement(path) {
+        const { argument } = path.node;
+
+        if (argument) {
+          ({ value, meta: updatedMeta } = traverseExpressionWithEvaluation(argument, meta));
+        }
+
+        path.stop();
+      },
+    });
+  } else {
+    ({ value, meta: updatedMeta } = traverseExpressionWithEvaluation(expression.body, meta));
+  }
+
+  return createResultPair(value as t.Expression, updatedMeta);
+};
+
+/**
+ * Will look in an expression and return the actual value along with updated metadata.
+ * If the expression is an identifier node (a variable) and a constant,
+ * it will return the variable reference.
+ *
+ * E.g: If there was a identifier called `color` that is set somewhere as `const color = 'blue'`,
+ * passing the `color` identifier to this function would return `'blue'`.
+ *
+ * This behaviour is the same for const string & numeric literals,
+ * and object expressions.
+ *
+ * @param expression Expression we want to interrogate.
+ * @param state Babel state - should house options and meta data used during the transformation.
+ */
+export const traverseExpressionWithEvaluation = (
```

Comment:
nit: can we shorten this function name?

`evaluateExpression` is short and sweet.

Reply:
hehehhee nice. `evaluateExpression`, works for me.

> There are only two hard things in Computer Science: cache invalidation and naming things.
-- Phil Karlton

😛